test/Npgsql.Tests/Support/PgServerMock.cs

```diff
@@ -0,0 +1,228 @@
+
+            foreach (var field in fields)
+            {
+                _writeBuffer.WriteNullTerminatedString(field.Name);
+                _writeBuffer.WriteUInt32(field.TableOID);
+                _writeBuffer.WriteInt16(field.ColumnAttributeNumber);
+                _writeBuffer.WriteUInt32(field.TypeOID);
+                _writeBuffer.WriteInt16(field.TypeSize);
+                _writeBuffer.WriteInt32(field.TypeModifier);
+                _writeBuffer.WriteInt16((short)field.FormatCode);
+            }
+
+            return this;
+        }
+
+        internal PgServerMock WriteDataRow(params byte[][] columnValues)
+        {
+            CheckDisposed();
+
+            _writeBuffer.WriteByte((byte)BackendMessageCode.DataRow);
+            _writeBuffer.WriteInt32(4 + 2 + columnValues.Sum(v => 4 + v.Length));
+            _writeBuffer.WriteInt16(columnValues.Length);
+
+            foreach (var field in columnValues)
+            {
+                _writeBuffer.WriteInt32(field.Length);
+                _writeBuffer.WriteBytes(field);
+            }
+
+            return this;
+        }
+
+        internal PgServerMock WriteCommandComplete(string tag = "")
+        {
+            CheckDisposed();
+
+            _writeBuffer.WriteByte((byte)BackendMessageCode.CommandComplete);
+            _writeBuffer.WriteInt32(4 + Encoding.GetByteCount(tag) + 1);
+            _writeBuffer.WriteNullTerminatedString(tag);
+            return this;
+        }
+
+        internal PgServerMock WriteReadyForQuery(TransactionStatus transactionStatus = TransactionStatus.Idle)
+        {
+            CheckDisposed();
+            _writeBuffer.WriteByte((byte)BackendMessageCode.ReadyForQuery);
+            _writeBuffer.WriteInt32(4 + 1);
+            _writeBuffer.WriteByte((byte)transactionStatus);
+            return this;
+        }
+
+        internal PgServerMock WriteAuthenticateOk()
+        {
+            CheckDisposed();
+            _writeBuffer.WriteByte((byte)BackendMessageCode.AuthenticationRequest);
+            _writeBuffer.WriteInt32(4 + 4);
+            _writeBuffer.WriteInt32(0);
+            return this;
+        }
+
+        internal PgServerMock WriteParameterStatuses(Dictionary<string, string> parameters)
+        {
+            foreach (var kv in parameters)
+                WriteParameterStatus(kv.Key, kv.Value);
+            return this;
+        }
+
+        internal PgServerMock WriteParameterStatus(string name, string value)
+        {
+            CheckDisposed();
+
+            _writeBuffer.WriteByte((byte)BackendMessageCode.ParameterStatus);
+            _writeBuffer.WriteInt32(4 + Encoding.GetByteCount(name) + 1 + Encoding.GetByteCount(value) + 1);
+            _writeBuffer.WriteNullTerminatedString(name);
+            _writeBuffer.WriteNullTerminatedString(value);
+
+            return this;
+        }
+
+        internal PgServerMock WriteBackendKeyData(int processId, int secret)
```

Comment:
I was taking another look, and it seems we don't use `processId `and `secret` params here. Is there a reason for this?

Reply:
Of course you're right - this was a mistake.